### PR TITLE
Balder/use templated find take 2

### PR DESCRIPTION
**Pull request by baldersheim**
@vekterli PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@vekterli Ping

**Review by vekterli — APPROVED** [review]
:+1: a couple minor comments. Biggest difference between the Working Group proposal and this is that heterogenous equality comparisons are enabled by default instead of only being enabled when a heterogenous hashing type is used. Since we don't have any particular issues related to backwards compatibility this is probably not a problem.

File: searchlib/src/vespa/searchlib/aggregation/group.cpp [hunk 1]
```diff
@@ -103,7 +103,7 @@ Group::Value::groupSingle(const ResultNode & selectResult, HitRank rank, const G
     }
     GroupHash & childMap = *_childInfo._childMap;
     Group * group(NULL);
-    GroupHash::iterator found = childMap.find<ResultNode, GroupResult, ResultHash, ResultEqual>(selectResult, GroupResult(&_children));
+    GroupHash::iterator found = childMap.find<ResultNode>(selectResult);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/aggregation/group.cpp]: no, fixed

File: searchlib/src/vespa/searchlib/fef/properties.cpp [hunk 2]
```diff
@@ -209,7 +209,7 @@ Properties::lookup(vespalib::stringref key) const
     if (key.empty()) {
         return Property();
     }
-    Map::const_iterator node = _data.find(key);
+    Map::const_iterator node = _data.find<vespalib::stringref>(key);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/fef/properties.cpp]: No, fixed

File: staging_vespalib/src/vespa/vespalib/objects/identifiable.cpp [hunk 3]
```diff
@@ -31,14 +31,30 @@ class Register {
...
+        uint32_t operator() (const RuntimeClass * f) const { return hashValue(f->name()); }
+        uint32_t operator() (const char * name) const { return hashValue(name); }
+
+    };
+    struct EqualName {
+        bool operator() (const RuntimeClass * a, const RuntimeClass * b) const { return strcmp(a->name(), b->name()) == 0; }
+        bool operator() (const RuntimeClass * a, const char * b) const { return strcmp(a->name(), b) == 0; }
+        bool operator() (const char * a, const RuntimeClass * b) const { return strcmp(a, b->name()) == 0; }
+
+    };
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply staging_vespalib/src/vespa/vespalib/objects/identifiable.cpp]: The dragons comes at night..

**Pull request merged**